File: scratch/read_pdf.py
import sys

try:
    import pypdf
    reader = pypdf.PdfReader(r"D:\onepws\imperrial\Imperial Essence Website Requirements Response Updated.pdf")
    for i, page in enumerate(reader.pages):
        print(f"--- PAGE {i+1} ---")
        print(page.extract_text())
except Exception as e:
    print("pypdf error:", e)
    try:
        import pdfplumber
        with pdfplumber.open(r"D:\onepws\imperrial\Imperial Essence Website Requirements Response Updated.pdf") as pdf:
            for i, page in enumerate(pdf.pages):
                print(f"--- PAGE {i+1} ---")
                print(page.extract_text())
    except Exception as e2:
        print("pdfplumber error:", e2)

File: scratch/read_pdf.js
const fs = require('fs');

const pdfPath = 'D:\\onepws\\imperrial\\Imperial Essence Website Requirements Response Updated.pdf';
const buf = fs.readFileSync(pdfPath);
const content = buf.toString('latin1');

// Extract readable text chunks between parenthesis in PDF streams
const matches = content.match(/\(([^()]{2,})\)/g) || [];
const textBlocks = matches
  .map(m => m.slice(1, -1))
  .filter(t => t.trim().length > 0 && !/^[\d\s\.\,\-\/\:\;\#\%\=\+\*\!\?]+$/.test(t));

console.log('--- ALL EXTRACTED TEXT FROM PDF ---');
console.log([...new Set(textBlocks)].join('\n'));
